presto-hive/src/main/java/io/prestosql/plugin/hive/security/SemiTransactionalHiveSqlStandardAccessControlMetadataMetastore.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive.security;
+
+import io.prestosql.plugin.hive.authentication.HiveIdentity;
+import io.prestosql.plugin.hive.metastore.HivePrincipal;
+import io.prestosql.plugin.hive.metastore.HivePrivilegeInfo;
+import io.prestosql.plugin.hive.metastore.SemiTransactionalHiveMetastore;
+import io.prestosql.spi.security.RoleGrant;
+
+import java.util.Optional;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+public class SemiTransactionalHiveSqlStandardAccessControlMetadataMetastore
+        implements SqlStandardAccessControlMetadataMetastore
+{
+    private final SemiTransactionalHiveMetastore hiveMetastore;
```

Comment:
That might be silly, but did you consider making `SemiTransactionalHiveMetastore` implement `SqlStandardAccessControlMetadataMetastore` directly?

Reply:
Hmm - good point :)

Reply:
It may be a bit constraining if we want to change structure of it. But we can postpone adding an extra delegation layer.